state/watcher/watcher.go

```diff
@@ -109,12 +114,16 @@ var Period time.Duration = 5 * time.Second
 
 // New returns a new Watcher observing the changelog collection,
 // which must be a capped collection maintained by mgo/txn.
-func New(changelog *mgo.Collection) *Watcher {
+func New(changelog *mgo.Collection, iteratorFunc func() mongo.Iterator) *Watcher {
 	w := &Watcher{
-		log:     changelog,
-		watches: make(map[watchKey][]watchInfo),
-		current: make(map[watchKey]int64),
-		request: make(chan interface{}),
+		log:          changelog,
+		iteratorFunc: iteratorFunc,
+		watches:      make(map[watchKey][]watchInfo),
+		current:      make(map[watchKey]int64),
+		request:      make(chan interface{}),
+	}
+	if w.iteratorFunc == nil {
```

Comment:
We use this pattern elsewhere. But we also use a test New function also in other places. Matter of taste as to how one does DI. Doing it this way makes the dependency explicit. I can go either way.